src/Forms/PresentedOptionsField.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace SilverStripe\CKANRegistry\Forms;
+
+use SilverStripe\CKANRegistry\Model\Resource;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\TextField;
+
+/**
+ * A PresentedOptionsField renders either a list of options that can be chosen for a {@link DropdownFilter}, or
+ * a text area allowing free text entry per line.
+ *
+ * The values of these options are serialised and saved as JSON.
+ */
+class PresentedOptionsField extends TextField
+{
+    /**
+     * @var int
+     */
+    const SELECT_TYPE_ALL = 0;
+
+    /**
+     * @var int
+     */
+    const SELECT_TYPE_CUSTOM = 1;
+
+    /**
+     * @var int
+     */
+    const SELECT_TYPE_DEFAULT = self::SELECT_TYPE_ALL;
+
+    protected $schemaComponent = 'PresentedOptions';
+
+    /**
+     * The resource that this options field will suggest options from
+     *
+     * @var Resource
+     */
+    protected $resouce;
+
+    public function __construct(
+        $name,
+        Resource $resource,
+        $title = null,
+        $value = '',
+        $maxLength = null,
+        Form $form = null
+    ) {
+        $this->setResouce($resource);
+
+        parent::__construct($name, $title, $value, $maxLength, $form);
+
+        $this->addExtraClass('ckan-presented-options__container');
+    }
+
+    public function Type()
+    {
+        return 'ckan-presented-options';
+    }
+
+    public function getSchemaDataDefaults()
+    {
+        $data = parent::getSchemaDataDefaults();
+        $data['data']['options'] = self::getOptions();
+        $data['data']['selectTypeDefault'] = self::SELECT_TYPE_DEFAULT;
+        $data['data']['selectTypes'] = self::getSelectTypes();
+        $data['data']['endpoint'] = $this->getResouce()->Endpoint;
+        $data['data']['resource'] = $this->getResouce()->Identifier;
+        $data['data']['fieldMap'] = $this->getResouce()->Fields()->map('ID', 'Name')->toArray();
```

Comment:
We could probably reduce the repetition of `$data['data']['` by using `array_merge()` instead